### PR TITLE
Implemented new style BTD in picmi

### PR DESCRIPTION
**Pull request by dpgrote**
In `picmi.LabFrameFieldDiagnostic`, added the option `warpx_new_BTD`. When true, the new method of BTD is used, allowing the use of the extra parameters, like `warpx_format` and `warpx_openpmd_backend`.

This will become the default method when the old method is removed. Similar changes will be made to `picmi.LabFrameParticleDiagnostic` when the new method can write out particles.

### CODE REVIEW

File: Python/pywarpx/picmi.py [hunk 1]
```diff
@@ -1104,6 +1121,62 @@ def initialize_inputs(self):
...
+                fields_to_plot.add('Ez')
+            elif dataname == 'B':
+                fields_to_plot.add('Bx')
+                fields_to_plot.add('By')
+                fields_to_plot.add('Bz')
+            elif dataname == 'J':
+                fields_to_plot.add('jx')
+                fields_to_plot.add('jy')
+                fields_to_plot.add('jz')
+            elif dataname in ['Ex', 'Ey', 'Ez', 'Bx', 'By', 'Bz', 'rho']:
```
Comment (ax3l) [inline Python/pywarpx/picmi.py:1175]: Additionally, we have `rho_<species>` now :)

File: Python/pywarpx/picmi.py [hunk 2]
```diff
@@ -1092,7 +1092,24 @@ def initialize_inputs(self):
 
 
 class LabFrameFieldDiagnostic(picmistandard.PICMI_LabFrameFieldDiagnostic):
+    def init(self, kw):
+        self.use_new_BTD = kw.pop('warpx_new_BTD', False)
```
Comment (ax3l) [inline Python/pywarpx/picmi.py:1107]: Should we add a docstring `"""..."""` to describe the kwargs that are possible to be passed here?

**Review by ax3l — COMMENTED** [review]
Cool!

File: Python/pywarpx/picmi.py (hunk 1 above)
Comment (dpgrote) [reply Python/pywarpx/picmi.py:1175]: Nice! I didn't see it. Now added.

File: Python/pywarpx/picmi.py (hunk 2 above)
Comment (dpgrote) [reply Python/pywarpx/picmi.py:1107]: Good idea, though it has to be a bit kludgy to retain the full doc string. I added it for this case. If this looks good, a later PR can add the doc for the rest of the classes.

**Review by ax3l — APPROVED** [review]
Thanks a lot! :tada:

**Pull request merged**